packages/server-wallet/src/protocols/process-ledger-queue.ts

```diff
@@ -55,17 +55,15 @@ import {
 
 export type ProtocolState = {
   fundingChannel: ChannelStateWithSupported;
-  theirLedgerProposal?: SimpleAllocation;
-  myLedgerProposal?: SimpleAllocation;
-  theirLedgerProposalNonce: number;
-  myLedgerProposalNonce: number;
+  theirLedgerProposal: {proposal: SimpleAllocation | null; nonce: number};
+  myLedgerProposal: {proposal: SimpleAllocation | null; nonce: number};
   channelsRequestingFunds: ChannelState[];
   channelsReturningFunds: ChannelState[];
 };
 
 type ProtocolStateWithCommits = ProtocolState & {
-  theirLedgerProposal: SimpleAllocation;
-  myLedgerProposal: SimpleAllocation;
+  theirLedgerProposal: {proposal?: SimpleAllocation; nonce: number};
+  myLedgerProposal: {proposal?: SimpleAllocation; nonce: number};
 };
```

Comment:
The types here are confusing.
- Sometimes `proposal` must be present, but it may have the `null` value
- Other times, `proposal` may be `undefined`

It smells bad that the presence of a `null` value contains different information than the absence of the key.

Reply:
You're right, this is a typo — this type was meant to have definitely not null or undefined types